### ISSUE TITLE
Fails to run on Mac SDL2 dependency and OpenGL

### ISSUE THREAD

**Issue by topcheese** [Larpon/shy#6]
I forgot the exact command, but I think it's -glcore33 or something like it that I have to add in order to use any Sokol related libraries.   I'm using an older Mac from 2011, so you may not want to support it.
I also had to do a "brew install pkg-config sdl2" for the first error.

Here is the last error.
Assertion failed: (desc->context.metal.device), function _sg_mtl_setup_backend, file sokol_gfx.h, line 10752.

**Comment by Larpon** [comment]
Yes macOS is not very well tested yet. My own mac is very old and I do not have access to other hardware so mac development is slow.

Linux, Windows and Android is what I have the best access to currently so development is leaning towards those platforms. I know that all dependencies *should* work under mac, including using metal for rendering - it just needs time

**Comment by topcheese** [comment]
Yes my MBP is from and 2011, so anything above that should support metal. I'll keep this mind when testing your framework.  Thanks for the response and your work!

**Comment by Larpon** [comment]
I've just commited [dbfa590](https://github.com/Larpon/shy/commit/dbfa5908a58f455228b5089a93d840606fe1fe49).

This makes it so you can do e.g. `v run  ~/.vmodules/shy/examples/image/image.v` without specifying any `-d` flags, and thus use OpenGL core 3.3 on macOS per default.

I think for now it is best to to make `metal` opt-in via `-d metal`. Please not that I haven't had time to actually implement proper Metal support. But it is on my (long) TODO list :sweat_smile:

**Comment by Larpon** [comment]
Running my test project [`puzzle_vibes`](https://github.com/Larpon/puzzle_vibes) works and isn't too demanding:
![image](https://github.com/Larpon/shy/assets/768942/c6e67ae3-428d-4c82-8bed-9b94373233d4)
